network/src/server.rs

```diff
@@ -0,0 +1,42 @@
+use crate::constants;
+use async_std::{net::TcpStream, prelude::*};
+use std::{
+    convert::{TryFrom, TryInto},
+    error::Error,
+    sync::Arc,
+};
+
+/// Call this method on the server side on accepted `stream` connection
+/// with the proper `request_handler` to answer request messages
+pub async fn handle_client<T>(
+    request_handler: Arc<T>,
+    stream: &mut TcpStream,
```

Comment:
The purpose was to make an abstraction over `TcpStream` - usage of it as a parameter will not solve this problem.